tests/kubevirt_helper.go

```diff
@@ -477,24 +485,163 @@ func AreVolumeReplicasCollocated(vol *volume.Volume, globalReplicSet []*api.Repl
+					Name:      "kubevirt-creds",
+					Namespace: "default",
+				},
+				Data: map[string]string{
+					"fio-vm-multi-disk": "ubuntu",
+				},
+			}
+			_, err = k8sCore.CreateConfigMap(configMap)
+			if err != nil {
+				log.Infof("Failed to create config map kubevirt-creds: %v", err)
+				return err
+			}
+			log.Infof("Created config map kubevirt-creds")
+		} else {
+			// An unexpected error occurred when retrieving the ConfigMap
+			log.Infof("Failed to retrieve config map kubevirt-creds: %v", err)
+			return err
+		}
+	} else {
+		log.Infof("Config map kubevirt-creds already exists")
+	}
+	return nil
+}
+
+// WriteFilesAndStoreMD5InVM write few files in the VM and calculates the md5sum
+func WriteFilesAndStoreMD5InVM(virtualMachines []*scheduler.Context, namespace string, fileCount int, maxFileSize int) error {
+	log.Infof("Creating %d files and storing their MD5 checksums in namespace %s", fileCount, namespace)
+	createFilesCmd := fmt.Sprintf("mkdir -p ~/testfiles && cd ~/testfiles && "+
+		"rm -f ~/file_checksums.md5 && for i in $(seq 1 %d); do "+
+		"head -c $(($RANDOM %% %d)) </dev/urandom >file$i; md5sum file$i >> ~/file_checksums.md5; done", fileCount, maxFileSize)
+	for _, appCtx := range virtualMachines {
+		vms, err := GetAllVMsFromScheduledContexts([]*scheduler.Context{appCtx})
+		if err != nil {
+			return err
+		}
+		for _, v := range vms {
+			_, err := RunCmdInVM(v, createFilesCmd, context1.TODO())
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// ValidateFileIntegrityInVM validates the md5sum of files that we wrote in the VM
+func ValidateFileIntegrityInVM(virtualMachines []*scheduler.Context, namespace string) error {
+	log.Infof("Validating file integrity in namespace %s", namespace)
+	validateFilesCmd := "cd ~/testfiles && md5sum -c ~/file_checksums.md5"
+	for _, appCtx := range virtualMachines {
+		vms, err := GetAllVMsFromScheduledContexts([]*scheduler.Context{appCtx})
+		if err != nil {
+			return err
+		}
+		for _, v := range vms {
+			_, err := RunCmdInVM(v, validateFilesCmd, context1.TODO())
+			if err != nil {
+				return err
+			}
 		}
-		_, err = core.Instance().CreateConfigMap(configMap)
+	}
+	return nil
+}
+
+// ListEvents lists all events in a namespace in logs.
+func ListEvents(namespace string) error {
+	eventList, err := k8sCore.ListEvents(namespace, metav1.ListOptions{})
+	if err != nil {
+		log.Infof("Failed to list events in namespace %s: %v", namespace, err)
+		return err
+	}
+	log.Infof("Events in namespace %s:", namespace)
+	for _, event := range eventList.Items {
+		log.Infof("Time: %v, Event: %s, Type: %s, Reason: %s, Object: %s/%s, Message: %s",
+			event.FirstTimestamp, event.Name, event.Type, event.Reason, event.InvolvedObject.Kind, event.InvolvedObject.Name, event.Message)
+	}
+	return nil
+}
+
+// HotAddPVCsToKubevirtVM hot adds disk to a running VM
+func HotAddPVCsToKubevirtVM(virtualMachines []*scheduler.Context, numberOfDisks int, size string) error {
```

Comment:
Can we make use of AddPVCsToVirtualMachin? We may need to include a Hotpluggable parameter.